File: Go/백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

func main() {
	defer bw.Flush()
	bs.Split(bufio.ScanWords)

	N := input()
	arr := make([]int, N)
	for i := 0; i < N; i++ {
		arr[i] = input()
	}
	ans := 0
	asc := 1
	dsc := 1
	for i := 1; i < N; i++ {
		if arr[i] > arr[i-1] {
			asc++
			ans = max(ans, dsc)
			dsc = 1
		} else if arr[i] < arr[i-1] {
			dsc++
			ans = max(ans, asc)
			asc = 1
		}
	}
    
	ans = max(ans, asc, dsc)
	bw.WriteString(strconv.Itoa(ans))
}

var (
	bs = bufio.NewScanner(os.Stdin)
	bw = bufio.NewWriter(os.Stdout)
)

func input() int {
	bs.Scan()
	num, _ := strconv.Atoi(bs.Text())
	return num
}

File: Python/백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.py
N = int(input())
if N == 1: exit(print(1))
arr = [*map(int,input().split())]
res = [0] * (N-1)
for i in range(N-1):
    comp = arr[i] - arr[i+1]

    if comp > 0:
        if i > 0 and res[i-1] > 0:
            res[i] = res[i-1] + 1
        else:
            res[i] = 1
    elif comp < 0:
        if i > 0 and res[i-1] < 0:
            res[i] = res[i-1] - 1
        else:
            res[i] = -1
    else:
        res[i] = res[i-1]

print(max(abs(x) + 1 for x in res))
